confignode/src/main/java/org/apache/iotdb/confignode/manager/load/balancer/router/LoadScoreGreedyRouter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.confignode.manager.load.balancer.router;
+
+import org.apache.iotdb.common.rpc.thrift.TConsensusGroupId;
+import org.apache.iotdb.common.rpc.thrift.TDataNodeLocation;
+import org.apache.iotdb.common.rpc.thrift.TRegionReplicaSet;
+import org.apache.iotdb.tsfile.utils.Pair;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Vector;
+import java.util.concurrent.ConcurrentHashMap;
+
+/** The LoadScoreGreedyRouter always pick the Replica with the lowest loadScore */
+public class LoadScoreGreedyRouter implements IRouter {
+
+  // Map<DataNodeId, loadScore>
+  private final Map<Integer, Float> loadScoreMap;
```

Comment:
Suggest define a class such as `LoadScore` to replace the raw `Float` here. 

Even if we only consider the Float here, we may have more scores from different dimension to calculate the raw score of DataNode

Reply:
Good idea! But the LoadScoreGreedyRouter is only sorted based on the loadScore, then greedily select DataNode with the lowest loadScore, where using a Float should be sufficient. I think it would be more appropriate to create the LoadScore class later when we extending other function classes that implement the IRouter interface.